### PR TITLE
[FTAMS102020] CeydaUlubas

### PR DESCRIPTION
**Pull request by ceydaulubas**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,26 +1,90 @@
...
+import contacts from './contacts.json'
+
+const App = () => {
+  //Iteration 1 | Display 5 Contacts
+  const fiveContacts = contacts.slice(0, 5);
+  const [contactsState, setcontactsState] = useState(fiveContacts);
+
+  //Iteration 2 | Add New Random Contacts
+  const addRandomContacts = () => {
+    const randomContacts = Math.floor(Math.random() * contacts.length);
```
Comment (wax-bear) [inline src/App.js:12]: A random contact is added. However, this code results in duplicate contacts. :) 
Perhaps, we can check whether a contact is added before, or splice the contact from the contacts array when added.

File: src/App.js [hunk 2]
```diff
@@ -1,26 +1,90 @@
...
+      if (a.popularity > b.popularity)
+        return 1;
+      return 0;
+
+    });
+    setcontactsState(contactsStateCopy);
+  }
+
+  //Iteration 4 | Remove Contacts
+  const removeContacts = (id) => {
```
Comment (wax-bear) [inline src/App.js:46]: removeContacts works as expected, great!

File: src/App.js [hunk 3]
```diff
@@ -1,26 +1,90 @@
...
+        <table>
+          <tr>
+            <th>Picture</th>
+            <th>Name</th>
+            <th>Popularity</th>
+            <th>Action</th>
+          </tr>
+          {contactsState.map(item => (
+
+            <tr>
```
Comment (wax-bear) [inline src/App.js:74]: Please give this `tr` a key property with the id as the value, as required by React (check the console).

File: src/App.js [hunk 4]
```diff
@@ -1,26 +1,90 @@
...
+
+      <div className="buttonsContainer">
+        <button onClick={() => addRandomContacts()}>Add Random Contact</button>
+        <button onClick={() => sortByName()}>Sort by name</button>
+        <button onClick={() => sortByPopularity()}>Sort by popularity</button>
+      </div>
+
+      <div className="tableofContacts">
+        <table>
+          <tr>
```
Comment (wax-bear) [inline src/App.js:66]: This `tr` can be nested in a `thead`.

File: src/App.js [hunk 5]
```diff
@@ -1,26 +1,90 @@
...
+            <th>Picture</th>
+            <th>Name</th>
+            <th>Popularity</th>
+            <th>Action</th>
+          </tr>
+          {contactsState.map(item => (
```
Comment (wax-bear) [inline src/App.js:72]: These `tr`s can be nested in a `tbody`.

File: src/App.js [hunk 6]
```diff
@@ -1,26 +1,90 @@
...
+    contactsStateCopy.sort(function (a, b) {
+      if (a.name < b.name) { return -1; }
+      if (a.name > b.name) { return 1; }
+      return 0;
+    });
+    setcontactsState(contactsStateCopy);
+  }
+
+  //Iteration 3 b | Sort Contacts By Popularity
+  const sortByPopularity = () => {
```
Comment (wax-bear) [inline src/App.js:31]: The popularity is sorted ascending, is this what you would like to happen?

**Review by wax-bear — COMMENTED** [review]
Great stuff Ceyda!
Please check my comments for potential improvements.

**Comment by ceydaulubas** [comment]
Thomas, thank you for your feedback. I fixed the problem in iteartion 1.

File: src/App.js (hunk 6 above)
Comment (ceydaulubas) [reply src/App.js:31]: Yes, I wanted to see popularity from the lowest to the highest number.

**Pull request closed without merging**